packages/react/src/hooks/useKPIToken.ts

```diff
@@ -22,19 +22,21 @@ export function useKPIToken(params?: KPITokenParams): {
     useEffect(() => {
         let cancelled = false;
         async function fetchData(): Promise<void> {
-            if (!chain || !params?.kpiTokenAddress) return;
+            if (!chain || !params?.kpiTokenAddress || !!kpiToken) return;
```

Comment:
Let's imagine for a moment that the parameters change but there's a cached `kpiToken` in the state already. What would happen? I'm not 100% sure this scenario can happen in the first place, but maybe we should make this check a bit more robust, like: `kpiToken?.address === params?.address`?

Reply:
Either that, or maybe let's remove the check completely to avoid the potentially infinite loop since `kpiToken` is now both set in this hook and in its dependencies array.

Reply:
Not having that condition would cause an infinite loop since the `kpiToken` gets updated in the `useEffect` (this cause a re-render and then the `useEffect` gets executed again); I think we could improve the check as you suggest.

Reply:
Fixed in 05a8f6f94bc220a79a4167c080b959f95c02b496.